### PR TITLE
feat: install dashmate with deb package

### PR DESCRIPTION
**Pull request by shumkov**
## Issue being fixed or feature implemented

Currently, you can setup the hp masternode only providing a branch from which dashmate will be built. This approach is pretty convenient when you actively developing/testing dashmate but to slow form common cases. 

## What was done?

- Introduced `dashmate_version` variable to setup a node with dashmate deb package

## How Has This Been Tested?

Manually

## Breaking Changes

None

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added or updated relevant unit/integration/functional/e2e tests
- [x] I have made corresponding changes to the documentation

**For repository code-owners and collaborators only**
- [x] I have assigned this pull request to a milestone

### CODE REVIEW

**Comment by strophy** [comment]
Let's split up `ansible/roles/dashmate/tasks/main.yml` to call separate files for `dashmate_version` and `dashmate_branch` workflows? This way we have all the messy dependency installation stuff in one file, and can skip it with a single condition

**Comment by strophy** [comment]
Also need to add `dashmate_helper_image` to the config generator

File: ansible/roles/dashmate/tasks/main.yml [hunk 1]
```diff
@@ -231,52 +247,75 @@
...
+
+- name: Get package architecture
+  ansible.builtin.command: dpkg --print-architecture
+  register: deb_architecture
+  when: dashmate_version is defined
+  changed_when: false
+
+- name: Install dashmate package
+  ansible.builtin.apt:
+    deb: "{{ release_info.json.assets | selectattr('name', 'search', '_' + deb_architecture.stdout + '.deb') | map(attribute='browser_download_url') | first }}"
```
Comment (strophy) [inline ansible/roles/dashmate/tasks/main.yml:269]: This fails on arm64 with:
```
TASK [dashmate : Install dashmate package] *********************************************************************************************************************
fatal: [hp-masternode-3]: FAILED! => 
  msg: |-
    The task includes an option with an undefined variable. The error was: No first item, sequence was empty.. No first item, sequence was empty.
  
    The error appears to be in '/home/strophy/dash-network-deploy/ansible/roles/dashmate/tasks/main.yml': line 267, column 3, but may
    be elsewhere in the file depending on the exact syntax problem.
  
    The offending line appears to be:
  
  
    - name: Install dashmate package
      ^ here
```

**Comment by strophy** [comment]
The two amd64 nodes which successfully installed dashmate errored with:
```
fatal: [hp-masternode-2]: FAILED! => changed=false 
  cmd:
  - dashmate
  - update
  - --verbose
  delta: '0:00:01.605351'
  end: '2023-04-22 06:14:57.767148'
  msg: non-zero return code
  rc: 1
  start: '2023-04-22 06:14:56.161797'
  stderr: |2-
        InvalidConfigFileFormatError: Invalid
        '/home/dashmate/.dashmate/config.json' config format:
        config/dashmate/helper must NOT have additional properties
  stderr_lines: <omitted>
  stdout: ''
  stdout_lines: <omitted>
```

File: ansible/roles/dashmate/tasks/main.yml (hunk 1 above)
Comment (shumkov) [reply ansible/roles/dashmate/tasks/main.yml:269]: That's wired. You probably use an old version that doesn't have arm64 package yet. https://github.com/dashpay/platform/releases/tag/v0.24.0-dev.23

**Comment by shumkov** [comment]
The same problem. You use the old version of dashmate.

**Review by strophy — APPROVED** [review]
Tested working, great job mate :rocket:

**Pull request merged**